### DISCUSSION TITLE
Manipulating BVTKNodes via Blender python

### DISCUSSION THREAD

**Discussion by ThisNameNotUsed** [tkeskita/BVtkNodes#84] [Q&A] [answered]
Hello, where in Blender Python's bpy object are the BVTK nodes in a running scene? I currently have to read in thousands of VTK PolyData models and gather stats about the generated meshes from blender functions.

I am having a very hard time figuring out how to access BVTKNodes and manipulate their fields from Blender's python text editor.  I could really use two lines of working python examples to:
1. Access and update a field of a BVTKNode such as the FileName of a vtkPolyDataReader
2.  Force Update

The current documentation states:

It is possible to interact with nodes and live VTK objects via Blender’s Python Console. Python Console includes three help operators for BVTKNodes:

Get Node operator inserts text which returns access to active node.
Get VTK Object inserts command which returns access to VTK object of the active node.
Get Node Output inserts text which returns the Output of VTK object.

But I can't for the life of me find where those functions are in the bpy object while scripting in Blender's text editor.  I do believe two working lines of code would do the BVTK documentation a great service.

i.e.:
bpy.function.,path.to.node_tree["vtkPolyDataReader"].FilePath = "C:/(pathToVTKFile)"

and

bpy.fucntion.path.to.BVTK.updatescene()

**Comment by vigeesh** [comment, marked as answer]
Perhaps, you may want to have a look at #2 . This was done a couple of years ago. BVtkNodes has changed substantially since then, so it is possible that the example script (+ [updating part](https://github.com/tkeskita/BVtkNodes/issues/2#issuecomment-541307183)) may not work as intended.
Checking the live Python console info panel helped me in figuring out how to access/edit object properties.

**Reply by ThisNameNotUsed** [reply]
Thank you. That tip helped out so much!

As someone who learned Blender Python scripting just for BVTK nodes I think a few lines of sample code will REALLY benefit the BVTK nodes documentation. I absolutely could not find how to play with the nodes and links using the standard Blender Python documentation and online tutorials. Are they in a different spot than traditional nodes? Maybe I just had an unlucky bout of researching but the following few things would have saved me a lot of frustration if they were in the BVTK nodes docs:


`
    # nodes and links live here
    nodes = bpy.data.node_groups["NodeTree"].nodes 

    links = bpy.data.node_groups["NodeTree"].links

    # update like so
    nodes["vtkPolyDataReader"].m_FileName = yourFilePath

    links.new(nodes["vtkPolyDataReader"].outputs[0], nodes["VTK To Blender Mesh"].inputs[0]) 

    # to hit the 'update' button on the "VTK To Blender Mesh" node you can set something to True that this already True:
    nodes["VTK To Blender Mesh"].create_faces = True #this line hits the BVTK update button

    # If you're changing a few models in a for loop you may need to render each for the changes to take place to each:
    bpy.ops.wm.redraw_timer(type='DRAW_WIN_SWAP', iterations=1) #Blender will not redraw or actually create the mesh without this line. All the dimensions reported will be the same of whatever model was loaded last.
`

That last one is probably superfluous but I don't know..

**Comment by tkeskita** [comment]
For anyone finding this thread: I've now added a section to the docs about [Python Scripting BVTKNodes in Blender](https://bvtknodes.readthedocs.io/en/latest/BVTKNodes.html#scripting-in-blender).
